Panic at startup with 1.36.1 with snowflake driver

### Relevant telegraf.conf

```toml
Not sure which config triggers this.
```

### Logs from Telegraf

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10312caf0]

goroutine 1 [running]:
regexp.(*Regexp).ReplaceAllString(0x0, {0x14001392500, 0x40}, {0x109cf4482, 0xa})
    regexp/regexp.go:575 +0x50
github.com/snowflakedb/gosnowflake.maskClientSecret(...)
    github.com/snowflakedb/gosnowflake@v1.16.0/secret_detector.go:56
github.com/snowflakedb/gosnowflake.maskSecrets({0x14001392500?, 0xe?})
    github.com/snowflakedb/gosnowflake@v1.16.0/secret_detector.go:67 +0x44
github.com/snowflakedb/gosnowflake.(*sfTextFormatter).Format(0x14000a158c0, 0x14000f32540)
    github.com/snowflakedb/gosnowflake@v1.16.0/log.go:71 +0x28
github.com/sirupsen/logrus.(*Entry).write(0x14000f32540)
    github.com/sirupsen/logrus@v1.9.3/entry.go:289 +0xc8
github.com/sirupsen/logrus.(*Entry).log(0x14000f324d0, 0x4, {0x14001392500, 0x40})
    github.com/sirupsen/logrus@v1.9.3/entry.go:252 +0x3dc
github.com/sirupsen/logrus.(*Entry).Log(0x14000f324d0, 0x4, {0x1400109fb38?, 0x1?, 0x1?})
    github.com/sirupsen/logrus@v1.9.3/entry.go:304 +0x60
github.com/sirupsen/logrus.(*Entry).Logf(0x14000f324d0, 0x4, {0x109ec57c7?, 0x10?}, {0x1400127cbe0?, 0x1089ca370?, 0x140004e8cd8?})
    github.com/sirupsen/logrus@v1.9.3/entry.go:349 +0x84
github.com/sirupsen/logrus.(*Logger).Logf(0x14000c58900, 0x4, {0x109ec57c7, 0x35}, {0x1400127cbe0, 0x1, 0x1})
    github.com/sirupsen/logrus@v1.9.3/logger.go:154 +0x74
github.com/sirupsen/logrus.(*Logger).Infof(...)
    github.com/sirupsen/logrus@v1.9.3/logger.go:168
github.com/snowflakedb/gosnowflake.(*defaultLogger).Infof(0x140004e8cc0, {0x109ec57c7, 0x35}, {0x1400127cbe0, 0x1, 0x1})
    github.com/snowflakedb/gosnowflake@v1.16.0/log.go:192 +0x5c
github.com/snowflakedb/gosnowflake.newCrlCacheCleaner()
    github.com/snowflakedb/gosnowflake@v1.16.0/crl.go:97 +0x24c
github.com/snowflakedb/gosnowflake.init()
    github.com/snowflakedb/gosnowflake@v1.16.0/crl.go:57 +0x470
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x102cecaf0]
```

### System info

Telegraf 1.36.1

### Docker

_No response_

### Steps to reproduce

1. Launch
2. Crash
3.
...


### Expected behavior

Not crash :)

### Actual behavior

Panic

### Additional info

_No response_

--- a/go.mod
+++ b/go.mod
@@ -194,7 +194,7 @@ require (
 	github.com/sijms/go-ora/v2 v2.9.0
 	github.com/sirupsen/logrus v1.9.3
 	github.com/sleepinggenius2/gosmi v0.4.4
-	github.com/snowflakedb/gosnowflake v1.16.0
+	github.com/snowflakedb/gosnowflake v0.0.0-20250911095445-20c4d105d9a0
 	github.com/srebhan/cborquery v1.0.4
 	github.com/srebhan/protobufquery v1.0.4
 	github.com/stretchr/testify v1.11.1

